scipy/stats/_entropy.py

```diff
@@ -319,6 +319,8 @@ class of statistics based on spacings. Scandinavian Journal of
     """
     xp = array_namespace(values)
     values = xp.asarray(values)
+    if xp.isdtype(values.dtype, "integral"):
```

Comment:
```suggestion
    if xp.isdtype(values.dtype, "integral"):  # type: ignore[union-attr]
```

Reply:
why is MyPy unhappy here?

Reply:
I don't know. Whatever it is, it's wrong.
`values` is typed as `array_like`. Perhaps MyPy doesn't realize it's an array at this point.